modules/engage-paella-player/src/main/java/org/opencastproject/engage/paella/PaellaConfigRest.java

```diff
@@ -79,9 +83,16 @@ protected void setSecurityService(SecurityService securityService) {
   }
 
   public void activate(ComponentContext cc) {
-    logger.debug("activate()");
-
     paellaConfigFolder = cc.getBundleContext().getProperty(PAELLA_CONFIG_FOLDER_PROPERTY);
+
+    // Fall back to default location if necessary
+    if (StringUtils.isBlank(paellaConfigFolder)) {
+      paellaConfigFolder = cc.getBundleContext().getProperty("karaf.etc");
+      if (StringUtils.isBlank(paellaConfigFolder)) {
+        throw new ConfigurationException("Paella configuration not set and unable to fall back to default location");
+      }
+      paellaConfigFolder += "/paella";
```

Comment:
Hmm... Using a platform-indenpendent method that uses the correct path separator would be preferable, wouldn't it?

Reply:
Interesting. For me, this is the simpler and cleaner variant. The alternative does make the code more complex. But will push a patch in a minute.